packages/framework-provider-azure/src/helpers/query-helper.ts

```diff
@@ -214,3 +218,83 @@ function toLocalSortFor(
   })
   return sortedList
 }
+
+function buildProjections(projections: ProjectionFor<unknown> | string = '*'): string {
+  if (typeof projections !== 'object') {
+    return projections
+  }
+
+  // Group fields by the root property
+  const groupedFields: any = {}
+  Object.values(projections).forEach((field: string) => {
+    const root: string = field.split('.')[0]
+    if (!groupedFields[root]) {
+      groupedFields[root] = []
+    }
+    groupedFields[root].push(field)
+  })
+
+  return Object.keys(groupedFields)
+    .map((root: string): string => {
+      const fields = groupedFields[root]
+      if (root.endsWith('[]')) {
+        const arrayRoot: string = root.slice(0, -2)
+        const subFields = fields.map((f: string) => f.replace(`${root}.`, 'item.')).join(', ')
+        return `ARRAY(SELECT ${subFields} FROM item IN c.${arrayRoot}) AS ${arrayRoot}`
```

Comment:
Out-of-curiosity question: Who adds the `[]` to the fields path? Is it GraphQL?

Another question: What would happen if the array field is in the middle of the path? for example: `lastPurchase.items[].id`? Or is this something not supported?

Reply:
@alvaroloes the `[]` is added by the `getFields` method in `GraphQLGenerator` (see [here](https://github.com/Optum/booster/blob/09d7db60d6eb540e98936a40b94870b1d50b61ec/packages/framework-core/src/services/graphql/graphql-generator.ts#L257)).

Array fields in the middle of the path were not initially supported but were added in commit 8726201. However, only array fields that are part of a top-level object property are supported for selecting properties. Arrays nested deeper are not supported at the moment (in other words, `foo.items[].bar` is supported but not `foo.bar.items[].somethingElse`)

Reply:
Perfect, thanks!